packages/gatsby/src/utils/adapter/types.ts

```diff
@@ -149,6 +155,10 @@ export interface IAdapterConfig {
    * plugin and adapter is used at the same time.
    */
   pluginsToDisable?: Array<string>
+  /**
+   * TODO: write description
```

Comment:
just to add some context here what this new field here is (at least in current state of things):

adapters should provide absolute path to a commonjs (at least with current implementation due to usage of `require()` in `gatsby-plugin-utils`) file.

That file should have default export that implements a function with specific signature - `ImageCdnUrlGeneratorFn` type (imported from `gatsby` inside of adapter package) defines arguments and return types. There is also `ImageCdnSourceImage` and `ImageCdnTransformArgs` types for first and second params of that function - `packages/gatsby-adapter-netlify/src/image-cdn-url-generator.ts` is example of that module in this PR for Netlify (and those types are used there)